h2o-genmodel/src/test/java/hex/genmodel/algos/tree/TreeSHAPTest.java

```diff
@@ -0,0 +1,131 @@
+package hex.genmodel.algos.tree;
+
+import ai.h2o.algos.tree.INode;
+import ai.h2o.algos.tree.INodeStat;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class TreeSHAPTest {
+
+    @Test
+    public void calculateContributionsWithAlmostZeroWeights_modelApp() {
+        checkContributionsBehaveSmoothlyWithZeroWeights(new Model_App_TreeBuilder());
+    }
+
+    @Test
+    public void calculateContributionsWithAlmostZeroWeights_modelAppp() {
+        checkContributionsBehaveSmoothlyWithZeroWeights(new Model_Appp_TreeBuilder());
+    }
+
+    private void checkContributionsBehaveSmoothlyWithZeroWeights(TreeBuilder tb) {
+        final float closeToZeroWeight = 0.000001f;
+        TreeSHAP<String, Node, Node> zeroTreeSHAP = new TreeSHAP<>(tb.buildTree(0));
+        TreeSHAP<String, Node, Node> closeToZeroTreeSHAP = new TreeSHAP<>(tb.buildTree(closeToZeroWeight));
+        final int nContribs = 3;
+        String[] paths = {"LL", "LR", "RL", "RR"};
+
+        for (String path : paths) {
+            float[] expectedContribs = closeToZeroTreeSHAP.calculateContributions(path, new float[nContribs]);
+            float[] actualContribs   = zeroTreeSHAP.calculateContributions(path, new float[nContribs]);
+            assertArrayEquals("Contributions for path '" + path + "' should be similar",
+                    expectedContribs, actualContribs, 1e-2f);
+        }
+    }
+
+    private static class Model_App_TreeBuilder implements TreeBuilder { // Model A'' from support ticket #99203 (private)
+        @Override
+        public Node[] buildTree(float almostZeroWeight) {
+            float regularWeight = 0.5f - almostZeroWeight;
+            return new Node[]{
+                    new Node(1, 2, 1f, 0, 40),
+                    new Node(3, 4, 1f - (2 * almostZeroWeight), 1, 0),
+                    new Node(5, 6, almostZeroWeight * 2, 1, 40),
+                    new Node(regularWeight, 0),
+                    new Node(regularWeight, 0),
+                    new Node(almostZeroWeight, 0),
+                    new Node(almostZeroWeight, 80)
+            };
+        }
+    }
+
+    private static class Model_Appp_TreeBuilder implements TreeBuilder { // Model A''' from support ticket #99203 (private)
```

Comment:
What is the difference between Model_App and Model_Appp, what is means?

Reply:
thank you for this comment - I will address it in a new PR